app/Http/Controllers/PostController.php

```diff
@@ -39,13 +40,32 @@ public function store(Request $request)
     {
         $this->validate($request,[
             'title'  => 'required',
-            'body' => 'required'
+            'body' => 'required',
+            'cover_image' => 'image|nullable|max:1999'
         ]);
+        
+         //handle the file upload
+         if($request->hasFile('cover_image')){
+            //get filename with extension
+            $filenameWithExt = $request->file('cover_image')->getClientOriginalName();
+            //get just filename
+            $filename = pathinfo($filenameWithExt, PATHINFO_FILENAME);
+            //get just ext
+            $extension = $request->file('cover_image')->getClientOriginalExtension();
+            //filename to store
+            $fileNameToStore = $filename .'_'.time().'.'.$extension;
+            //upload image
+            $path = $request->file('cover_image')->storeAs('public/cover_images',$fileNameToStore);
```

Comment:
I have doubts with respect image driver yet 
why I need create a driver?
Why can not I use the default driver?

Reply:
The default is optional, you can use it if you want, you can use a custom driver for customize the location where you locate the storage files
After all you must to run the command storage link, this will create a "linked" folder on public